### DISCUSSION TITLE
Use % literally in search

### DISCUSSION THREAD

**Discussion by KasaiJo** [pocketbase/pocketbase#4704] [Q&A] [answered]
If I start from this base, no problem.

![image](https://github.com/pocketbase/pocketbase/assets/26089497/d21a536c-5f55-49ca-ba53-162b7b2d2c6b)

Obviously, if I add '%' (code987%), nothing is found since '%' is used as a wildcard. So we have a "begins with" expression.
But when I escape '%' I still can't find a result. What did I miss?

![image](https://github.com/pocketbase/pocketbase/assets/26089497/abb80205-080e-45fb-a429-9e9b00f2cd2a)

I've also tried a few solutions through the api, without success.

```js
    let escapedSearchValue = searchCode.value.replace('%', '\%');
    
    const records = await pb.collection('stock_item').getFullList({
      filter: `code~'${escapedSearchValue}'`,
      sort: '-created',
    });
```

Inspired by [this post.](https://github.com/pocketbase/pocketbase/discussions/1495)
```js
    const records = await pb.collection('stock_item').getFullList({
      filter: pb.filter('code ~ {:code}', { code: searchCode.value}),
      sort: '-created',
    });
```

**Comment by ganigeorgiev** [comment, marked as answer]
Hm, this seems like an issue with the auto `%` wrapping.

Escaping the `%` character with a backslash should work but the auto wildcard (wrapping with `%`) search is applied only if the string doesn't contain `%`. You can try with:

```
code ~ '%code\%%'
```

I'll investigate it in more details this weekend and will consider adding an exception for the escaped `%` and `_` characters.

**Reply by KasaiJo** [reply]
Thank you for quick response 👍 
I'll use your solution in the meantime.

**Reply by ganigeorgiev** [reply]
_Just for info in case someone stumble here - I've pushed a small fix in `master` (_aka. both `%code\%%` and `code\%` should behave the same_) and it will be available with the next minor v0.22.8 release sometime later this weekend._
